csrc/executor.cpp

```diff
@@ -1444,12 +1769,14 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
 
   // only allocate outputs when not given
   if (outputs.empty()) {
+    auto expr_eval = executor_utils::bindInputs(args, lowered_->kernel());
```

Comment:
This is still not really clear to me, but we also have `PrecomputedValues`, and in other parts of the executor, we use it to instantiate an `ExpressionEvaluator`. See for example `inferOutputSizes`. Should we do the same?

Reply:
We are not consistently always using the `PrecomputedValues` to instantiate an  `ExpressionEvaluator`. It's actually the opposite. In the same file, we are using `executor_utils::bindInputs` more often than the `PrecomputedValues`. I think this makes sense because there is only one instance of `PrecomputedValues`, and you need to bind inputs to that. I tried changing all of the to use `PrecomputedValues`, but seems that we are trying to bind different values to `PrecomputedValues`, which causes failure.

Reply:
Yeah, that's right. Maybe PrecomputedValues::bindInputs` should be done with a copy so that we could reuse it for different inputs.

Sounds like one more cleanup but not for this PR.